LibreHardwareMonitor/UI/PlotPanel.cs

```diff
@@ -73,24 +75,50 @@ private ContextMenu CreateMenu()
-                new MenuItem("20 min", (s, e) => { _timeAxis.Zoom(0, 20 * 60); InvalidatePlot(); }),
-                new MenuItem("30 min", (s, e) => { _timeAxis.Zoom(0, 30 * 60); InvalidatePlot(); }),
-                new MenuItem("45 min", (s, e) => { _timeAxis.Zoom(0, 45 * 60); InvalidatePlot(); }),
-                new MenuItem("1 h", (s, e) => { _timeAxis.Zoom(0, 60 * 60); InvalidatePlot(); }),
-                new MenuItem("1.5 h", (s, e) => { _timeAxis.Zoom(0, 1.5 * 60 * 60); InvalidatePlot(); }),
-                new MenuItem("2 h", (s, e) => { _timeAxis.Zoom(0, 2 * 60 * 60); InvalidatePlot(); }),
-                new MenuItem("3 h", (s, e) => { _timeAxis.Zoom(0, 3 * 60 * 60); InvalidatePlot(); }),
-                new MenuItem("6 h", (s, e) => { _timeAxis.Zoom(0, 6 * 60 * 60); InvalidatePlot(); }),
-                new MenuItem("12 h", (s, e) => { _timeAxis.Zoom(0, 12 * 60 * 60); InvalidatePlot(); }),
-                new MenuItem("24 h", (s, e) => { _timeAxis.Zoom(0, 24 * 60 * 60); InvalidatePlot(); }) };
-            foreach (MenuItem mi in timeWindowMenuItems)
-                timeWindow.MenuItems.Add(mi);
-            menu.MenuItems.Add(timeWindow);
+            MenuItem showAxesMenuItem = new MenuItem("Show Axes");
+            _showAxes = new UserOption("showAxes", true, showAxesMenuItem, _settings);
+            _showAxes.Changed += (sender, e) =>
+            {
+                if (_showAxes.Value)
+                    _model.PlotMargins = new OxyThickness(double.NaN);
+                else
+                    _model.PlotMargins = new OxyThickness(0);
+            };
+            menu.MenuItems.Add(showAxesMenuItem);
+
+            MenuItem timeAxisMenuItem = new MenuItem("Time Axis");
+            MenuItem[] timeAxisMenuItems =
+                { new MenuItem("Enable Zoom"),
+                new MenuItem("Auto", (s, e) => { TimeAxisZoom(0, double.NaN); }),
+                new MenuItem("5 min", (s, e) => { TimeAxisZoom(0, 5 * 60); }),
+                new MenuItem("10 min", (s, e) => { TimeAxisZoom(0, 10 * 60); }),
+                new MenuItem("20 min", (s, e) => { TimeAxisZoom(0, 20 * 60); }),
+                new MenuItem("30 min", (s, e) => { TimeAxisZoom(0, 30 * 60); }),
+                new MenuItem("45 min", (s, e) => { TimeAxisZoom(0, 45 * 60); }),
+                new MenuItem("1 h", (s, e) => { TimeAxisZoom(0, 60 * 60); }),
+                new MenuItem("1.5 h", (s, e) => { TimeAxisZoom(0, 1.5 * 60 * 60); }),
+                new MenuItem("2 h", (s, e) => { TimeAxisZoom(0, 2 * 60 * 60); }),
+                new MenuItem("3 h", (s, e) => { TimeAxisZoom(0, 3 * 60 * 60); }),
+                new MenuItem("6 h", (s, e) => { TimeAxisZoom(0, 6 * 60 * 60); }),
+                new MenuItem("12 h", (s, e) => { TimeAxisZoom(0, 12 * 60 * 60); }),
+                new MenuItem("24 h", (s, e) => { TimeAxisZoom(0, 24 * 60 * 60); }) };
+            foreach (MenuItem mi in timeAxisMenuItems)
```

Comment:
new line before for each for readability, also after the menu.MenuItems.Add below.

Reply:
OK. The original code had no new line before the foreach, but I agree it is better with it.